projects/driver_controls/test/test_ads1015.c

```diff
@@ -0,0 +1,60 @@
+#include <stdint.h>
+#include <stdio.h>
+#include "ads1015.h"
+#include "unity.h"
+
+#include "gpio.h"
+#include "gpio_it.h"
+#include "i2c.h"
+#include "i2c_mcu.h"
+#include "interrupt.h"
+#include "status.h"
+
+static ADS1015Data data;
+GPIOAddress ready_pin = { GPIO_PORT_B, 2 };
+
+static I2CSettings i2c_settings = {
+  .speed = I2C_SPEED_FAST, .scl = { GPIO_PORT_B, 10 }, .sda = { GPIO_PORT_B, 11 }
+};
+
+/*
+static void prv_callback(const GPIOAddress *address, void *context) {
+  ADS1015Data *data = context;
+  printf("converted from channel %d\n", data->current_channel);
+}
+*/
+
+void setup_test(void) {
+  gpio_init();
+  interrupt_init();
+  gpio_it_init();
+
+  i2c_init(I2C_PORT_2, &i2c_settings);
+
+  ads1015_init(&data, I2C_PORT_2, ADS1015_ADDRESS_GND, &ready_pin);
+}
+
+void teardown_test(void) {}
+
+void test_ads1015_init(void) {
```

Comment:
Try to make your test names more descriptive - for example, this seems more like a bounds/invalid input test for `ads1015_init`. (Which is a great test, by the way)